### ISSUE TITLE
bug: Ticket Classification UI starves the model of 11 of its 14 features, producing near-random predictions

### ISSUE THREAD

**Issue by iamvisheshsrivastava** [iamvisheshsrivastava/insightdesk-ai#16]
## Problem

Tested the live Ticket Classification page (https://insightdesk-ai-sqck.onrender.com) as an end user: submitted a clearly-worded ticket (\"Cannot log in to my account\" / \"I keep getting an invalid password error even though I am sure my password is correct. This started after the last update.\") and got back:

\`\`\`json
\"top_3_predictions\": {
  \"Feature Request\": 0.2059,
  \"Data Issue\": 0.2053,
  \"Security\": 0.2040
}
\`\`\`

Nearly uniform ~20% across every class — the model isn't discriminating at all, despite the ticket text being unambiguous.

## Root cause

The XGBoost feature pipeline (`src/features/feature_builder.py`) was trained on 14 columns:
- 8 numeric: `resolution_time_hours`, `previous_tickets`, `account_age_days`, `account_monthly_value`, `affected_users`, `ticket_text_length`, `response_count`, `attachments_count`
- 6 categorical: `priority`, `severity`, `customer_tier`, `product`, `product_module`, `region`
- plus TF-IDF text (`subject` + `description` concatenated)

`TicketInput` (`src/api/main.py:436-461`) gives all 8 numeric fields a fixed default (mostly `0`, `affected_users` defaults to `1`) and 3 of the categorical fields a fixed default (`customer_tier=\"standard\"`, `region=\"US\"`, `product_module=\"unknown\"`).

The live frontend's Ticket Classification form (`frontend/src/pages/TicketCategorization.jsx`) only collects `subject`, `description`, `error_logs`, `stack_trace`, `product`, `priority`, `severity` — it never asks for the other 8 fields. So every real submission through the UI sends the exact same constant values for 11 of the model's 14 trained features, and only the TF-IDF text signal actually varies. If the model learned to lean on the numeric/categorical business signals (very plausible for this kind of dataset — account value, ticket history, etc. are often more decisive than raw text for routing), starving it of those features explains the near-uniform, low-confidence output seen live.

## Suggested fix

Add the missing fields to the Ticket Classification form (with sensible defaults, e.g. a number input for `previous_tickets`/`affected_users`, a select for `customer_tier`/`region`/`product_module`) so real predictions get the model's full feature set. Alternatively, if these operational fields aren't realistic to collect from an end user in a support-ticket-submission flow, that's a signal the model itself may need retraining to rely more on text and less on fields a real intake form can't always supply — worth checking feature importances (`model.feature_importances_`) to see how much weight is actually on the TF-IDF block vs the rest.

**Comment by iamvisheshsrivastava** [comment]
Partly addressed in 322d8ea. Found that the UI showed 'Unknown' for real model output because it read `category` while the API returns `predicted_category`; fixed, plus real category colors. Also: ticket_text_length is now derived from the text, predictions under 0.35 confidence get `low_confidence` and a deterministic `keyword_suggestion`, the UI shows a warning, and the demo classifier now uses the real labels. Not done: adding the other numeric/categorical fields to the form, and retraining/feature-importance review (the training data classes look near-uniform, so low confidence may be inherent). Leaving open.
